subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/CleanupTestDirectoryExtension.groovy

```diff
@@ -42,10 +42,10 @@ class CleanupTestDirectoryExtension implements IAnnotationDrivenExtension<Cleanu
 
         @Override
         void intercept(IMethodInvocation invocation) throws Throwable {
+            TestDirectoryProvider provider = GroovyRuntimeUtil.getProperty(invocation.instance, fieldName) as TestDirectoryProvider
             def noCleanupOnErrorListener = new AbstractRunListener() {
```

Comment:
🤔 I wonder if the captured set could be reduced further by replacing this anonymous inner class with a static class.

Reply:
The `FailureCleanupInterceptor` is small enough, so, IMO, trading memory for readability is fine here.

Reply:
I see. No strong feeling either way.